source/Tooltip.h

```diff
@@ -0,0 +1,92 @@
+/* Tooltip.h
+Copyright (c) 2025 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "text/Alignment.h"
+#include "Rectangle.h"
+#include "text/WrappedText.h"
+
+#include <string>
+
+class Color;
+class Point;
+
+
+
+// A class for drawing the tooltips in a UI panel.
+class Tooltip {
+public:
+	// The direction from the draw point that the tooltip should be drawn in.
+	enum class Direction {
+		UP_LEFT,
+		UP_RIGHT,
+		DOWN_LEFT,
+		DOWN_RIGHT
+	};
+
+	// The corner from the drawn rectangle that the tooltip should be drawn from.
+	enum class Corner {
+		TOP_LEFT,
+		TOP_RIGHT,
+		BOTTOM_LEFT,
+		BOTTOM_RIGHT,
+	};
+
+	// The tooltip's state determines its background color.
+	enum class State {
+		NORMAL,
+		WARNING,
+		ERROR
+	};
```

Comment:
I'm not a fan of the shop tooltip colors being part of the Tooltip class, considering they're only used in that single place. Managing the colors should be a responsibility of the parent panel, with the tooltips having a function for setting an arbitrary color instead of a few hardcoded options.